engines/storage/postgres/castore.go

```diff
@@ -52,7 +52,11 @@ func (db *PostgresCAStore) SelectByType(ctx context.Context, CAType models.Certi
 }
 
 func (db *PostgresCAStore) SelectAll(ctx context.Context, req storage.StorageListRequest[models.CACertificate]) (string, error) {
-	return db.querier.SelectAll(ctx, req.QueryParams, []gormExtraOps{}, req.ExhaustiveRun, req.ApplyFunc)
+	opts := []gormExtraOps{
```

Comment:
The join to certificates is applied to all SelectAll calls, which may add unnecessary overhead when common-name filtering isn't used. Consider applying this join only when the request includes a common-name filter or scoping it to SelectByCommonName.